src/components/ComputeModal.js

```diff
@@ -678,18 +711,29 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
   const renderActionButton = () => {
     const { runtime: existingRuntime, hasGpu } = getExistingEnvironmentConfig()
     const { runtime: desiredRuntime } = getDesiredEnvironmentConfig()
-    const commonButtonProps = hasGpu && viewMode !== 'deleteEnvironmentOptions' ?
-      { disabled: true, tooltip: 'Cloud compute with GPU(s) cannot be updated. Please delete it and create a new one.' } :
-      { disabled: !hasChanges() || !!errors, tooltip: Utils.summarizeErrors(errors) }
-    const canShowCustomImageWarning = viewMode === undefined
+    const commonButtonProps = Utils.cond([
+      hasGpu && viewMode !== 'deleteEnvironmentOptions',
+      () => ({ disabled: true, tooltip: 'Cloud compute with GPU(s) cannot be updated. Please delete it and create a new one.' })
+    ], [
+      computeConfig.gpuEnabled && getValidGpusForZone(computeConfig.computeZone).length === 0 && viewMode !== 'deleteEnvironmentOptions',
```

Comment:
Lodash FP's `_.isEmpty()` may fit nicely here:

```suggestion
      computeConfig.gpuEnabled && _.isEmpty(getValidGpusForZone(computeConfig.computeZone)) && viewMode !== 'deleteEnvironmentOptions',
```

Reply:
Done